tests/integration/workflows/test_workflows_tasks_upload.py

```diff
@@ -65,3 +71,72 @@ def test_store_record_does_not_raise_in_the_orcid_receiver(mock_attempt_push, ap
         })
 
         store_record(obj, None)  # Does not raise.
+
+
+def test_store_root_new_record(isolated_app_with_commit):
+    config = {
+        'FEATURE_FLAG_ENABLE_MERGER': True
+    }
+
+    with patch.dict(current_app.config, config):
+        head = TestRecordMetadata.create_from_kwargs(index=False, has_pid=False)
+        head_uuid = head.record_metadata.id
+        record = head.record_metadata.json
+
+        obj = workflow_object_class.create(record)
+
+        root = {
+            'version': 'original',
+            'acquisition_source': {'source': 'arXiv'}
+        }
+
+        extra_data = {
+            'head_uuid': str(head_uuid),
+            'merger_root': root,
+        }
+
+        obj.extra_data = extra_data
+
+        store_root(obj, None)
+
+        root_entry = read_wf_record_source(head_uuid, 'arxiv')
+
+        assert root_entry.json == root
+
+
+def test_store_root_update_record(isolated_app_with_commit):
+    config = {
+        'FEATURE_FLAG_ENABLE_MERGER': True
+    }
+
+    with patch.dict(current_app.config, config):
+        head = TestRecordMetadata.create_from_kwargs(index=False, has_pid=False)
+        head_uuid = head.record_metadata.id
+        record = head.record_metadata.json
+
+        original_root = {
+            'version': 'original',
+            'acquisition_source': {'source': 'arXiv'},
+        }
+
+        update_root = {
+            'version': 'updated',
+            'acquisition_source': {'source': 'arXiv'},
+        }
+
+        insert_wf_record_source(json=original_root, record_uuid=head_uuid, source='arxiv')
+
+        obj = workflow_object_class.create(record)
+
+        extra_data = {
+            'head_uuid': str(head_uuid),
+            'merger_root': update_root,
+        }
+
+        obj.extra_data = extra_data
+
+        store_root(obj, None)
```

Comment:
also we can check that the ``merger_root`` key exists

Reply:
I feel like the logic in this test is sort of duplicated in the test_arxiv_merge.py . Should I remove it there?